daemon/loop.go

```diff
@@ -72,13 +50,22 @@ func (d *Daemon) GitPollLoop(stop chan struct{}, wg *sync.WaitGroup, logger log.
 		k(logger)
 	}
 
+	imagePollTimer := time.NewTimer(d.RegistryPollInterval)
+
 	// Ask for a sync, and to poll images, straight away
 	d.askForSync()
+	d.askForImagePoll()
 	for {
 		select {
 		case <-stop:
 			logger.Log("stopping", "true")
 			return
+		case <-d.pollImagesSoon:
+			d.pollForNewImages(logger)
+			imagePollTimer.Stop()
+			imagePollTimer = time.NewTimer(d.RegistryPollInterval)
```

Comment:
Yeah, and in fact in the previous incarnation it should have been a timer too, since it had the same treatment -- I must have had Ticker/Timer blindness.